### ISSUE TITLE
Bootstrap v4 — not working inside .col-lg-N

### ISSUE THREAD

**Issue by f1nnix** [xremix/xGallerify#16]
Hi, Toni, thanks for sharing such a great plugin.

Trying to use it with Bootstrap v4. If a gallery is rendered inside `.col-lg-N` column, all the photos are stacked.

``` html
<div class="row">
  <div class="col-lg-6 col-md-6 col-sm-6">
  </div>
  <div class="col-lg-6 col-md-6 col-sm-6">
    // gallery here
  </div>
</div>
```

**See demo: [https://jsfiddle.net/f1nn/e8qyq5g1/](https://jsfiddle.net/f1nn/e8qyq5g1/)**

I suppose it may be related to container minimum width. In other words, if gallery container is too narrow to xGallery, all the photos are just stacked.

Is there a way to render gallery regarless the width of outer container?

Thanks.

**Comment by xremix** [comment]
Hi f1nnix,

At first thanks for commenting, the great feedback and contributing to my gallery. At a first look this looks really like an issue. I haven't tried the gallery with bootstrap 4 yet but it is definitely my goal to get it running there. I'm currently taking a look from my mobile phone so I can't give you details. I'll try to take a look ASAP. 

I never had issues with bootstrap 3 yet, so it shouldn't be a big deal to fix it for bootstrap 4.

**Comment by xremix** [comment]
@f1nnix I think I found your issue. xGallerify has fixed breakpoints for the number on images that are going to be displayed, based on the outer container. 
There are different breakpoints variants that I call "modes". 

The "bootstrap" mode says it has the same breakpoints as the grid in bootstrap, but as your container is a col-sm-6 it does only show one image per row. If you have a really large monitor it should display more than one image per row. 
If you want to have multiple images in a row you can try the "small" mode for example. I just made a sample here, based on your code:

https://jsfiddle.net/epef0q13/4/

I can add additional modes if you tell me how many images you want at what width of the container. I hope this helps.

If you still have issues you can reopen the issue.

**Comment by xremix** [comment]
I updated the documentation based on your opened issue, I hope it clarifies a bit:
https://github.com/xremix/xGallerify#modes

**Comment by xremix** [comment]
I updated the documentation based on your opened issue @f1nnix , I hope it clarifies a bit:
https://github.com/xremix/xGallerify#modes

**Comment by f1nnix** [comment]
@xremix wow, awesome! Brilliant support, thank you very much! After you described how modes work, everything is clear.

Also, I do hope, it may be useful if we can specify custom breakpoint values. For example, while rendering, just check typeof `mode` property. If === `String`, use predefined template. Otherwise, the custom breakpoints are applied. The format may be simple:

``` js
{
  mode: [
    {
      maxWidth: 1440,
      count: 4
    },
    {
    maxWidth: 979,
    count: 3
    },
    ...
  ]
}
```

Great job, Toni!

**Comment by xremix** [comment]
That's a great idea @f1nnix, I thought briefly about something similar as well yesterday after your input. You idea with such a property is great, the single objects could have the properties maxWidth and minWidth. Depending on what is specified it could use a < or > character. Awesome idea again! It could also help me in having the internal code more clean.

**Comment by xremix** [comment]
@f1nnix I implemented custom modes with the new version v0.1.0 that you can see in #20
